flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/StandaloneResourceManager.java

```diff
@@ -92,6 +94,11 @@ protected void initialize() throws ResourceManagerException {
 		}
 	}
 
+	@VisibleForTesting
+	CompletableFuture<Boolean> isFailingUnfulfillableRequestAsync() {
+		return CompletableFuture.supplyAsync(this::isFailingUnfulfillableRequest, getMainThreadExecutor());
+	}
```

Comment:
I'd prefer to have these kind of methods in a `TestingStandaloneResourceManager` because then they don't clutter the production code.